lib/rubygems/core_ext/kernel_warn.rb

```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 # `uplevel` keyword argument of Kernel#warn is available since ruby 2.5.
-if RUBY_VERSION >= "2.5"
+if RUBY_VERSION >= "2.5" && !Gem::KERNEL_WARN_IGNORES_INTERNAL_ENTRIES
```

Comment:
Since the require of this file is already "protected", I think we don't need to check it again? If someone is loading this file directly for some reason, I guess we can let run the code?

Reply:
I prefer to be sure, i.e. that Kernel#warn is never overridden when `Gem::KERNEL_WARN_IGNORES_INTERNAL_ENTRIES` is a guarantee.
I see this as the main check and then the other check (`require 'rubygems/core_ext/kernel_warn' unless Gem::KERNEL_WARN_IGNORES_INTERNAL_ENTRIES`) is just an optimization.

Reply:
This way, even if some other part of RubyGems at some point in the future does `require 'rubygems/core_ext/kernel_warn'`, it will still only override `Kernel#warn` if needed, which seems more future-proof.

Reply:
I can't imagine how we could end up requiring the file from some other part of rubygems, honestly. But assuming that could happen, then I'd say let's remove the "optimization" on the require itself so we don't have carry it around everywhere. Checking the same thing right before & after a require feels weird, specially if there's an extra condition that you only check in one of the two places.

Reply:
OK, I removed the optimization condition, so the file is always loaded and all conditions are inside.